T1332

k88hudson opened the pull request:
Titlecard overlay plugin

ScottDowne committed:
[#434] Initial drag track events together.

ScottDowne committed:
[#434] multi drag and drop actually working now... Promise.

ScottDowne committed:
[#434] whitespace fixes and hasownproperty check.

ScottDowne committed:
[#434] Whitespace fixes.

ScottDowne committed:
[#434] caching of variables used in loops.

ScottDowne committed:
[#434] removed function param that was not needed, and variable name change.

ScottDowne committed:
[#434] added newline to eof.

ScottDowne committed:
[#434] added some comments.

ScottDowne committed:
[#434] Review fixes so far.

ScottDowne committed:
[#434] More review fixes, so far.

ScottDowne committed:
[34] Fixed a bug regarding revert, and drop not needing to return true/false anymore.

ScottDowne committed:
[#434] make sure droppable exists before dropping.

ScottDowne committed:
[#434] What I believe to be the last review fix.

ScottDowne committed:
[#434] Auto scrolling with multiple drags.

mjschranz committed:
Added Butter test runner [#1192]

ScottDowne committed:
[#434] rebased. fixed scrolling bug. Fixed selected events while track remove bug. fixed droppable to null on drop bug.

ScottDowne committed:
[#434] bug where a draggable can have two droppables fixed.

ScottDowne committed:
[#434] resolving merge conflict.

ScottDowne committed:
[#434] Finished rebase.

jbuck committed:
Fixed linting errors [#1297]

mjschranz committed:
[#1296] Adding multiple sources now no longer changes the source of the video to the newly added source

jbuck committed:
Fixed checking of CSRF token before page is loaded [#1293]

ScottDowne committed:
[#t434] Multiple droppables for each draggable being handled.

ScottDowne committed:
[#t434] Back to single droppables for each draggable being handled.

mjschranz committed:
[#1327] Added media-view.css to .gitignore

mjschranz committed:
[#1296] Fixed review nit

mjschranz committed:
[#1308] Auth Button is no longer clickable when logged in

Commit:
1. Rewritten Googlemap editor
2. Added support for Stamen Tiles
3. Live updates to tracks in Butter
4. No terrain or Stamen-Terrain - they are very unstable [#1271 #1257]

Commit:
Added Googlemap editor to default-config.json [#1271]

secretrobotron committed:
[t943] UA warning

secretrobotron committed:
[t943] review fix - height of warning

dseif committed:
Trackevents use plugin defaults [#1343]

humphd committed:
[t1350] Remove vimeo related manual-tests from 0.5.x

mjschranz committed:
[#1316] Template Configs can now specify a project ID and have butter load it from cornfield instead of only project JSON

Commit:
Fixes and performance enhances for Googlemap Editor [#t1271]

humphd committed:
Merge pull request #385 from humphd/t1350

[t1350] Remove vimeo related manual-tests from 0.5.x r=matt,brett

humphd committed:
[t1364] Don't run cornfield tests if server not running.

mjschranz committed:
[#1316] Fixed to simply add id/name of project in saved JSON to butter project on load

jbuck committed:
Merge pull request #373 from secretrobotron/t943

Added UA check on start-up [#943]

humphd committed:
Merge pull request #391 from humphd/t1296-v0.5.x

[t1296] Fix current source issue when multiple sources are added r=dseif,chris

Commit:
added all templates to template directory

Commit:
Review fixes to Googlemap Editor [#1271]

humphd committed:
Merge pull request #398 from humphd/t1316

[t1316] Unable to specify the user's project id when it is loading r=bobby,humph

Commit:
linted, standardized, cleaned up music video, context, and report

Commit:
fixed more lint errors, updated popup, removed unecessary files, moved jquery and jquery UI to shared folder

Commit:
reset robots and newscaster to base config and template.js

Commit:
added video overlay divs

humphd committed:
[t1374] Remove references to Vimeo in 0.5.x branch

mjschranz committed:
[#1317] Exported projects are not broken when templates originiated from iFrames

ScottDowne committed:
[#434] Whitespace fix.

mjschranz committed:
[#1317] Removed old reference to TEMPLATES from cornfield config

humphd committed:
Merge pull request #399 from humphd/t1374

[t1374] Remove references to Vimeo in 0.5.x branch r=dseif,matt

k88hudson commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7d20f0be7729d86/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/7d20f0be7729d86/output.txt

Total script time: 0.17 mins

k88hudson commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3ea66791d353392/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3ea66791d353392/output.txt

Total script time: 0.82 mins

jbuck committed:
Merge pull request #401 from mjschranz/t1317

T1317 Export HTML is not taking into consideration the domain

Commit:
1. Review changes
2. Fix the way stamen types are given ids in the map registry
3. Enabled the terrain types [#1271]

Commit:
added new popup videos

Commit:
added all templates into a supported folder updated config files

Commit:
fixed paths in popup editors

mjschranz committed:
[#1386] Fixed cornfield test failing still when run on node server

Commit:
fixed lint errors in popup editor:

Commit:
added multiple sources to context

Commit:
changed template videos to remote version

Commit:
Removed extra newline [#1271]

Commit:
added pop sounds

Commit:
style fixes

Commit:
removed footnote track

Commit:
removed footnote track

ScottDowne commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/27fff6b9cadbfbc/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/27fff6b9cadbfbc/output.txt

Total script time: 0.12 mins

Commit:
Merge branch 'v0.5.x' of git://github.com/mozilla/butter into v0.5.x

ScottDowne committed:
[#1381] Updating to newest popcorn.

Commit:
worked around lint complaining

Commit:
added check-templates to lint CSS and JS of templates dir, removed templates dir from check

Commit:
took out duplication

Commit:
added license

Commit:
abstracted js and css checks further

Commit:
removed random track names

Commit:
removed bottom and right properties from popup

Commit:
added defaults fixed global i problem in popup and words

Commit:
removed popcorn stuff from robots for now

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+  _teardown: function( options ) {
+    var target = options._target;
+    if ( target ) {
+    target.removeChild( options._container );
```
Whitespace!

ScottDowne replied:
Actually, it is alright to do:

target && target.removeChild();

here.

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+
+    media.parentNode.appendChild( ctxContainer );
+
+  return ctxContainer;
```
whitespace.

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+    about: {
+    name: "Popcorn titlecard Plugin",
+    version: "0.1",
+    author: "@k88hudson"
```
top three lines. Whitespace.

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+      container = options._container = document.createElement( "div" ),
+      h2 = options.title && "<h2>" + options.title + "</h2>\n" || "",
+      h5 = options.subheading &&  "<h5>" + options.subheading + "</h5>\n" || "",
+      titleString;
```
Top 5 lines should be aligned with target.

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+   * options variable
+   */
+  start: function( event, options ) {
+    options._container.classList.add("on");
```
Whitespace around "on"

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+      h5 = options.subheading &&  "<h5>" + options.subheading + "</h5>\n" || "",
+      titleString;
+
+    container.classList.add("title-card");
```
Whitespace around "title-card"

ScottDowne commented on templates/shared/plugins/titlecard/popcorn.titlecard.js:
```diff
@@ -0,0 +1,118 @@
+   * options variable
+   */
+  end: function( event, options ) {
+    options._container.classList.remove("on");
```
Same as above.

Commit:
removed some parameters on checkSS

jbuck committed:
Merge pull request #396 from k88hudson/v0.5.x

Added core templates [#1367]

Commit:
[TEMPLATES] added titlecard plugin

Commit:
fixed spacing

jbuck commented:
Resolved by pull request #416 (Toronto, represent.)
